### ISSUE TITLE
List is empty. java.util.NoSuchElementException: 

### ISSUE THREAD

**Issue by selectee** [viartemev/requestmapper#84]
IntelliJ IDEA 2017.3.5
Build #IU-173.4674.33, built on March 5, 2018

JRE: 1.8.0_152-release-1024-b15 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

Plugin version  0.11

```
List is empty.
java.util.NoSuchElementException: List is empty.
	at kotlin.collections.CollectionsKt___CollectionsKt.first(_Collections.kt:178)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.extractParameterNameWithType(SpringMappingAnnotation.kt:76)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.fetchMappingsFromMethod(SpringMappingAnnotation.kt:62)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.fetchRequestMappingItem(SpringMappingAnnotation.kt:22)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.values(SpringMappingAnnotation.kt:16)
	at com.viartemev.requestmapper.RequestMappingContributor$findRequestMappingItems$3.invoke(RequestMappingContributor.kt:40)
	at com.viartemev.requestmapper.RequestMappingContributor$findRequestMappingItems$3.invoke(RequestMappingContributor.kt:12)
	at kotlin.sequences.FlatteningSequence$iterator$1.ensureItemIterator(Sequences.kt:253)
	at kotlin.sequences.FlatteningSequence$iterator$1.hasNext(Sequences.kt:241)
	at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:654)
	at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:684)
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:675)
	at com.viartemev.requestmapper.RequestMappingContributor.findRequestMappingItems(RequestMappingContributor.kt:41)
	at com.viartemev.requestmapper.RequestMappingContributor.getNames(RequestMappingContributor.kt:18)
	at com.intellij.ide.util.gotoByName.ContributorsBasedGotoByModel$1.processInReadAction(ContributorsBasedGotoByModel.java:98)
	at com.intellij.ide.util.gotoByName.ContributorsBasedGotoByModel$1.processInReadAction(ContributorsBasedGotoByModel.java:82)
	at com.intellij.openapi.application.ReadActionProcessor.lambda$process$0(ReadActionProcessor.java:28)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:47)
	at com.intellij.openapi.application.ReadActionProcessor.process(ReadActionProcessor.java:28)
	at com.intellij.concurrency.JobLauncherImpl.a(JobLauncherImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.concurrency.JobLauncherImpl.a(JobLauncherImpl.java:111)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1122)
	at com.intellij.concurrency.JobLauncherImpl.a(JobLauncherImpl.java:122)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:57)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:57)
	at com.intellij.ide.util.gotoByName.ContributorsBasedGotoByModel.processNames(ContributorsBasedGotoByModel.java:120)
	at com.intellij.ide.util.gotoByName.DefaultChooseByNameItemProvider.filterElements(DefaultChooseByNameItemProvider.java:80)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.a(ChooseByNameBase.java:1440)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.a(ChooseByNameBase.java:1416)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.a(ChooseByNameBase.java:1398)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.performInReadAction(ChooseByNameBase.java:1362)
	at com.intellij.openapi.progress.util.ReadTask.a(ReadTask.java:66)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:47)
	at com.intellij.openapi.progress.util.ReadTask.runBackgroundProcess(ReadTask.java:66)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.runBackgroundProcess(ChooseByNameBase.java:1349)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.a(ProgressIndicatorUtils.java:238)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.a(ProgressIndicatorUtils.java:236)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.access$000(ProgressIndicatorUtils.java:47)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$4.run(ProgressIndicatorUtils.java:187)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


java.util.NoSuchElementException: List is empty.
	at kotlin.collections.CollectionsKt___CollectionsKt.first(_Collections.kt:178)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.extractParameterNameWithType(SpringMappingAnnotation.kt:76)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.fetchMappingsFromMethod(SpringMappingAnnotation.kt:62)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.fetchRequestMappingItem(SpringMappingAnnotation.kt:22)
	at com.viartemev.requestmapper.annotations.spring.SpringMappingAnnotation.values(SpringMappingAnnotation.kt:16)
	at com.viartemev.requestmapper.RequestMappingContributor$findRequestMappingItems$3.invoke(RequestMappingContributor.kt:40)
	at com.viartemev.requestmapper.RequestMappingContributor$findRequestMappingItems$3.invoke(RequestMappingContributor.kt:12)
	at kotlin.sequences.FlatteningSequence$iterator$1.ensureItemIterator(Sequences.kt:253)
	at kotlin.sequences.FlatteningSequence$iterator$1.hasNext(Sequences.kt:241)
	at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:654)
	at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:684)
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:675)
	at com.viartemev.requestmapper.RequestMappingContributor.findRequestMappingItems(RequestMappingContributor.kt:41)
	at com.viartemev.requestmapper.RequestMappingContributor.getNames(RequestMappingContributor.kt:18)
	at com.intellij.ide.util.gotoByName.ContributorsBasedGotoByModel$1.processInReadAction(ContributorsBasedGotoByModel.java:98)
	at com.intellij.ide.util.gotoByName.ContributorsBasedGotoByModel$1.processInReadAction(ContributorsBasedGotoByModel.java:82)
	at com.intellij.openapi.application.ReadActionProcessor.lambda$process$0(ReadActionProcessor.java:28)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:47)
	at com.intellij.openapi.application.ReadActionProcessor.process(ReadActionProcessor.java:28)
	at com.intellij.concurrency.JobLauncherImpl.a(JobLauncherImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.concurrency.JobLauncherImpl.a(JobLauncherImpl.java:111)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1122)
	at com.intellij.concurrency.JobLauncherImpl.a(JobLauncherImpl.java:122)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:57)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:57)
	at com.intellij.ide.util.gotoByName.ContributorsBasedGotoByModel.processNames(ContributorsBasedGotoByModel.java:120)
	at com.intellij.ide.util.gotoByName.DefaultChooseByNameItemProvider.filterElements(DefaultChooseByNameItemProvider.java:80)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.a(ChooseByNameBase.java:1440)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.a(ChooseByNameBase.java:1416)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.a(ChooseByNameBase.java:1402)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.performInReadAction(ChooseByNameBase.java:1362)
	at com.intellij.openapi.progress.util.ReadTask.a(ReadTask.java:66)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:47)
	at com.intellij.openapi.progress.util.ReadTask.runBackgroundProcess(ReadTask.java:66)
	at com.intellij.ide.util.gotoByName.ChooseByNameBase$CalcElementsThread.runBackgroundProcess(ChooseByNameBase.java:1349)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.a(ProgressIndicatorUtils.java:238)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.a(ProgressIndicatorUtils.java:236)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.access$000(ProgressIndicatorUtils.java:47)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$4.run(ProgressIndicatorUtils.java:187)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)



```

**Comment by viartemev** [comment]
@selectee could you add a spring method which was a cause of an error.

**Comment by selectee** [comment]
```java
@RequestMapping(value = "/openVesselAndSentAllAttachmentsToRoute/{sendingSessionId}", method = RequestMethod.POST)
```

**Comment by viartemev** [comment]
@selectee what about the signature of a method which was annotated by this annotation?

**Comment by viartemev** [comment]
@selectee anyway, it's a good catch, I will fix it. If you use path variables in the url without `@PathVariable` annotation it will work, but you can't fetch the value of the variable.

**Comment by selectee** [comment]
@viartemev sure I'm use `@PathVariable` annotation.

```java
public Result openVesselAndSentAllAttachmentsToRoute(
            @RequestBody RequestInfo request,
            @PathVariable("sendingSessionId") long sendingSessionId) {
```

**Comment by viartemev** [comment]
@selectee fixed, will be in release 0.12.0
